controllers/storageclaim_controller.go

```diff
@@ -613,3 +643,103 @@ func (r *StorageClassClaimReconciler) createOrReplaceVolumeSnapshotClass(volumeS
 	}
 	return nil
 }
+
+func (r *StorageClaimReconciler) migrateStorageClassClaims() error {
+	storageClassClaimsList := &v1alpha1.StorageClassClaimList{}
+	if err := r.list(storageClassClaimsList); err != nil {
+		return fmt.Errorf("failed to list storageclassclaims")
+	}
+
+	for idx := range storageClassClaimsList.Items {
+		storageClassClaim := &storageClassClaimsList.Items[idx]
+		storageClaim := &v1alpha1.StorageClaim{}
+		storageClaim.Name = storageClassClaim.Name
+
+		// migrate storageclassclaim to storageclaim
+		_, err := controllerutil.CreateOrUpdate(r.ctx, r.Client, storageClaim, func() error {
+			storageClaim.Spec.Type = storageClassClaim.Spec.Type
```

Comment:
We agreed to change `BlockPool` to `Block` and this copy does not reflect that decision

Reply:
> We agreed to change
- I don't remember any such agreement. No qualms w/ suggestion.

Reply:
I can search for the comment that raised it up if needed

Reply:
Sure, pls do, context can help as on second thought I think this also needs provoder side changes.

Reply:
No necessarily, you can always translate it back when sending it to the provider as a hack. But the point of StorageClassClaim was always to never describe the request in terms of Rook or Ceph to prevent leaking of information. But 'BlockPool' is a rook/Ceph and leaking.

Reply:
Will do.

Reply:
done.